### DISCUSSION TITLE
PermissionError: [WinError 5] Access is denied

### DISCUSSION THREAD

**Discussion by brendanb132** [TobiasPankner/Teams-Auto-Joiner#148] [Q&A]
Hi!

Running into an issue upon running auto_joiner.py

> Microsoft Windows [Version 10.0.19043.1237]
> (c) Microsoft Corporation. All rights reserved.
> 
> C:\Users\brend\Documents\Teams-Auto-Joiner-master\Teams-Auto-Joiner-master>python auto_joiner.py
> Enabled headless mode
> Traceback (most recent call last):
>   File "C:\Users\brend\Documents\Teams-Auto-Joiner-master\Teams-Auto-Joiner-master\auto_joiner.py", line 770, in <module>
>     main()
>   File "C:\Users\brend\Documents\Teams-Auto-Joiner-master\Teams-Auto-Joiner-master\auto_joiner.py", line 618, in main
>     init_browser()
>   File "C:\Users\brend\Documents\Teams-Auto-Joiner-master\Teams-Auto-Joiner-master\auto_joiner.py", line 180, in init_browser
>     browser = webdriver.Chrome(ChromeDriverManager().install(), options=chrome_options)
>   File "C:\Users\brend\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\LocalCache\local-packages\Python39\site-packages\webdriver_manager\chrome.py", line 23, in __init__
>     self.driver = ChromeDriver(name=name,
>   File "C:\Users\brend\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\LocalCache\local-packages\Python39\site-packages\webdriver_manager\driver.py", line 54, in __init__
>     self.browser_version = chrome_version(chrome_type)
>   File "C:\Users\brend\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\LocalCache\local-packages\Python39\site-packages\webdriver_manager\utils.py", line 139, in chrome_version
>     stdout = os.popen(cmd).read()
>   File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.9_3.9.2032.0_x64__qbz5n2kfra8p0\lib\os.py", line 983, in popen
>     proc = subprocess.Popen(cmd,
>   File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.9_3.9.2032.0_x64__qbz5n2kfra8p0\lib\subprocess.py", line 951, in __init__
>     self._execute_child(args, executable, preexec_fn, close_fds,
>   File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.9_3.9.2032.0_x64__qbz5n2kfra8p0\lib\subprocess.py", line 1420, in _execute_child
>     hp, ht, pid, tid = _winapi.CreateProcess(executable, args,
> PermissionError: [WinError 5] Access is denied

The only solution I could find online was to run python as an administrator, but that didn't seem to do the trick. Thoughts?

**Comment by TobiasPankner** [comment]
Seems to be an issue with downloading the webdriver. Have you tried without headless?

**Reply by brendanb132** [reply]
Just tried with headless marked as false, same result.

![image](https://user-images.githubusercontent.com/77910459/135100107-203dd4a0-1e17-4480-ae1b-8d9170182bb9.png)

**Comment by TobiasPankner** [comment]
And how did you run it as admin? Open cmd with admin rights?

**Reply by brendanb132** [reply]
Yep, cmd is being ran in admin mode.
![image](https://user-images.githubusercontent.com/77910459/135099884-bb577607-386e-4400-9dc5-c2acad66b421.png)

**Comment by TobiasPankner** [comment]
Okay, you can try downloading the webdriver manually (make sure its the correct version). After that replace "ChromeDriverManager.install()" with the path to the exe.

**Comment by TobiasPankner** [comment]
Actually, I have another idea. Put this `os.environ['WDM_LOCAL'] = '1'` at the start of main in the code.  
Line 604 of the newest version.

**Reply by brendanb132** [reply]
Actually I think I tracked down the issue to windows forcing the folder to be read-only (god knows why). Once I figure that out I have a hunch it will run just fine.

**Reply by TobiasPankner** [reply]
Okay, pretty weird. But yeah then that is the problem.
